atomate/vasp/firetasks/write_inputs.py

```diff
@@ -289,34 +290,49 @@ def run_task(self, fw_spec):
 
 
 @explicit_serialize
-class UpdateScanRelaxBandgap(FiretaskBase):
+class WriteScanRelaxFromPBE(FiretaskBase):
     """
     Writes input files for a SCAN relaxation by constructing a new input set.
     The purpose of this Firetask is to allow the KSPACING and smearing parameters
-    to be recalculated based on the bandgap from the PBE relaxation in the
+    to be recalculated based on the bandgap from the PBESol pre-optimization in the
     SCAN relaxation workflow. Assumes that output files from a previous
-    (e.g., optimization) run can be accessed in current dir or prev_calc_dir.
+    (e.g., optimization) run have been copied to the current directory.
+
+    Note that if the "bandgap" kwarg is set by the user, this value will override
+    the value estimated from the PBESol relaxation.
 
     Optional params (dict):
-        override_default_vasp_params: Dict of any keyword arguments supported
-                                      by MPScanRelaxSet.
+        vasp_input_set_params: Dict of any keyword arguments supported by MPScanRelaxSet.
         potcar_spec (bool): Instead of writing the POTCAR, write a
             "POTCAR.spec". This is intended to allow testing of workflows
             without requiring pseudo-potentials to be installed on the system.
 
     """
-    optional_params = ["override_default_vasp_params", "potcar_spec"]
+    optional_params = ["vasp_input_set_params", "potcar_spec"]
 
     def run_task(self, fw_spec):
 
-        kwargs = self.get("override_default_vasp_params")
         potcar_spec = self.get("potcar_spec", False)
+        vasp_input_set_params = self.get("vasp_input_set_params") or {}
+
+        # update the bandgap based on output from the previous calculation,
+        # unless the user specified a bandgap via vasp_input_set_params
+        if not vasp_input_set_params.get("bandgap"):
```

Comment:
If a bandgap of zero is passed, this will evaluate to `False`. I'm not sure if someone would actually do that but maybe change to:
```suggestion
        if vasp_input_set_params.get("bandgap") is None:
```